src/serac/numerics/functional/tuple.hpp

```diff
@@ -13,216 +13,32 @@
- * @tparam T3 The fourth type stored in the tuple
- * @tparam T4 The fifth type stored in the tuple
- * @tparam T5 The sixth type stored in the tuple
- */
-template <typename T0, typename T1, typename T2, typename T3, typename T4, typename T5>
-struct tuple<T0, T1, T2, T3, T4, T5> {
-  T0 v0;  ///< The first member of the tuple
-  T1 v1;  ///< The second member of the tuple
-  T2 v2;  ///< The third member of the tuple
-  T3 v3;  ///< The fourth member of the tuple
-  T4 v4;  ///< The fifth member of the tuple
-  T5 v5;  ///< The sixth member of the tuple
-};
-
-/**
- * @brief Type that mimics std::tuple
- *
- * @tparam T0 The first type stored in the tuple
- * @tparam T1 The second type stored in the tuple
- * @tparam T2 The third type stored in the tuple
- * @tparam T3 The fourth type stored in the tuple
- * @tparam T4 The fifth type stored in the tuple
- * @tparam T5 The sixth type stored in the tuple
- * @tparam T6 The seventh type stored in the tuple
- */
-template <typename T0, typename T1, typename T2, typename T3, typename T4, typename T5, typename T6>
-struct tuple<T0, T1, T2, T3, T4, T5, T6> {
-  T0 v0;  ///< The first member of the tuple
-  T1 v1;  ///< The second member of the tuple
-  T2 v2;  ///< The third member of the tuple
-  T3 v3;  ///< The fourth member of the tuple
-  T4 v4;  ///< The fifth member of the tuple
-  T5 v5;  ///< The sixth member of the tuple
-  T6 v6;  ///< The seventh member of the tuple
-};
-
-/**
- * @brief Type that mimics std::tuple
- *
- * @tparam T0 The first type stored in the tuple
- * @tparam T1 The second type stored in the tuple
- * @tparam T2 The third type stored in the tuple
- * @tparam T3 The fourth type stored in the tuple
- * @tparam T4 The fifth type stored in the tuple
- * @tparam T5 The sixth type stored in the tuple
- * @tparam T6 The seventh type stored in the tuple
- * @tparam T7 The eighth type stored in the tuple
- */
-template <typename T0, typename T1, typename T2, typename T3, typename T4, typename T5, typename T6, typename T7>
-struct tuple<T0, T1, T2, T3, T4, T5, T6, T7> {
-  T0 v0;  ///< The first member of the tuple
-  T1 v1;  ///< The second member of the tuple
-  T2 v2;  ///< The third member of the tuple
-  T3 v3;  ///< The fourth member of the tuple
-  T4 v4;  ///< The fifth member of the tuple
-  T5 v5;  ///< The sixth member of the tuple
-  T6 v6;  ///< The seventh member of the tuple
-  T7 v7;  ///< The eighth member of the tuple
-};
-
-/**
- * @brief Class template argument deduction rule for tuples
- * @tparam T The variadic template parameter for tuple types
- */
-template <typename... T>
-tuple(T...) -> tuple<T...>;
-
 /**
  * @brief helper function for combining a list of values into a tuple
  * @tparam T types of the values to be tuple-d
  * @param args the actual values to be put into a tuple
  */
 template <typename... T>
-SERAC_HOST_DEVICE tuple<T...> make_tuple(const T&... args)
+SERAC_HOST_DEVICE camp::tuple<T...> make_tuple(const T&... args)
```

Comment:
Can you add a comment to these as why they exist here instead of in camp?

Reply:
Done.